target/linux/layerscape/image/Makefile

```diff
@@ -27,8 +29,16 @@ define Build/ls-append-dtb
 	dd if=$(DTS_DIR)/$(1).dtb >> $@
 endef
 
+define Build/ls-append-kernel
+	mkdir -p $(STAGING_DIR)/kerneldir-tmp
```

Comment:
Will this directory be unique for each device?

I'd rather use something like
```
	mkdir -p $@.tmp
	cp $(IMAGE_KERNEL) $@.tmp/fitImage
	make_ext4fs -J -L kernel -l "$(LS_SD_KERNELPART_SIZE)M" "$@.kernel.part" "$@.tmp"
	dd if=$@.kernel.part >> $@
```

where the names are constructed from the "image name" $@, and thus inherently unique.

The idea is prototyped here:
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/image/generic.mk#L56

Reply:
BTW: The intermediate files should also be removed again.

Reply:
This is better approach. I reworked it.

Reply:
I think we should remove $@.kernel.part as well?
And probably it would be good to chain the commands like in the reference, i.e. add ` && \` after each line?

Reply:
Done.